File: SISPRESTAMOS -BackEnd/Sistema_Prestamo/src/main/java/com/sisprestamo/services/PrestamoService.java
package com.sisprestamo.services;

import java.util.List;
import java.util.Map;

import com.sisprestamo.entity.Prestamo;

public interface PrestamoService {
	
	public Prestamo registrar(Prestamo prestamo);
	
	List<Map<String, Object>> obtenerInformacionPrestamos(Long idUsuario, Long idPrestamo);
	List<Map<String, Object>> obtenerInformacionPrestamosJefePrestamista(Long idPrestamo);
	List<Map<String, Object>> spListarPrestamos(Long idUsuario);
	List<Map<String, Object>> spListarPrestamosJefePrestamista(Long idUsuario);
	void actualizarMontoPagado(Long idPrestamo , Double montoIngresado);
	void actualizarMoraPagada(Long idPrestamo , Double moraIngresada);
}

File: SISPRESTAMOS -FrontEnd/Sistema_Prestamos/src/app/services/prestamo.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class PrestamoService {
  private apiUrl = 'http://localhost:8091/api/prestamos';  

  constructor(private http: HttpClient) { }

  public obtenerInformacionPrestamos(id_usuario: number, id_prestamo: number): Observable<any> {
    return this.http.get(`${this.apiUrl}/obtenerInformacion?idUsuario=${id_usuario}&idPrestamo=${id_prestamo}`);
  }

  public obtenerInformacionPrestamosJefePrestamista(id_prestamo: number): Observable<any> {
    return this.http.get(`${this.apiUrl}/obtenerInformacionJefePrestamista?idPrestamo=${id_prestamo}`);
  }

  public listarPrestamos(id_usuario: number): Observable<any> {
    return this.http.get(`${this.apiUrl}/listar?idUsuario=${id_usuario}`);
  }

  public listarPrestamosPorJefePrestamista(id_usuario: number): Observable<any> {
    return this.http.get(`${this.apiUrl}/listarPorJefePrestamista?idUsuario=${id_usuario}`);
  }

  actualizarMontoPagado(idPrestamo: number, montoIngresado: number): Observable<any> {
    const url = `${this.apiUrl}/actualizarMontoPagado?idPrestamo=${idPrestamo}&montoIngresado=${montoIngresado}`;
    return this.http.put(url, null); 
  }

  actualizarMoraPagada(idPrestamo: number, moraIngresada: number): Observable<any> {
    const url = `${this.apiUrl}/actualizarMoraPagada?idPrestamo=${idPrestamo}&moraIngresada=${moraIngresada}`;
    return this.http.put(url, null); 
  }

}